pageserver/src/tenant/storage_layer.rs

```diff
@@ -318,7 +318,7 @@ pub(crate) struct LayerFringe {
 #[derive(Debug)]
 struct LayerKeyspace {
     layer: ReadableLayer,
-    target_keyspace: KeySpace,
+    target_keyspace: Vec<KeySpace>,
```

Comment:
nit: making this member a `KeySpaceRandomAccum` would have been more expressive in my opinion, but this is fine.